src/app/qgsmaptoolfillring.cpp

```diff
@@ -127,7 +128,10 @@ void QgsMapToolFillRing::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
       return;
     }
 
-    g = QgsGeometry::fromPolygonXY( QgsPolygonXY() << pointList );
+    QgsLineString ext( pointList );
+    QgsPolygon polygon;
```

Comment:
```suggestion
    std::unique_ptr< QgsPolygon > polygon;
    polygon->setExteriorRing( ext.clone() );
    g = QgsGeometry( std::move( polygon ) );
```

(Avoids an unnecessary copy)

Reply:
Right! Thanks.